couchdb/client.py

```diff
@@ -284,19 +284,26 @@ def verify_user(self, token_or_name, password=None):
         :return: True if authenticated ok
         :rtype: bool
         """
-        if password is None:
-            header = {
-                'Accept': 'application/json',
-                'Cookie': 'AuthSession=' + token_or_name,
-            }
-        else:
-            from string import strip
-            from base64 import encodestring
-            header = {
-                'Accept': 'application/json',
-                'Authorization': 'Basic ' + strip(encodestring(token_or_name + ':' + password)),
-            }
-        status, _, _ = self.resource.get_json('_session', headers=header)
+        from http import Unauthorized, ServerError
+        from string import strip
+        from base64 import encodestring
+        try:
+            if password is None:
+                header = {
+                    'Accept': 'application/json',
+                    'Cookie': 'AuthSession=' + token_or_name,
+                }
+                status, _, _ = self.resource.get_json('_session', headers=header)
+            else:
+                header = {
+                    'Accept': 'application/json',
+                    'Authorization': 'Basic ' + strip(encodestring(token_or_name + ':' + password)),
```

Comment:
If you're duplicating this logic in a few methods, perhaps you should extract some helper methods to create these header dicts?